src/Sarif/Baseline/ResultMatching/DataStructures/MatchedResults.cs

```diff
@@ -82,6 +84,9 @@ public Result CalculateNewBaselineResult(DictionaryMergeBehavior propertyBagMerg
             {
                 ResultMatchingProperties.Add(MatchedResults.MatchResultMetadata_RunKeyName, PreviousResult.OriginalRun.Id.InstanceGuid);
             }
+
+            LogicalLocations = PreviousResult.OriginalRun.LogicalLocations;
```

Comment:
>LogicalLocations [](start = 58, length = 16)

At first I worried that you were duplicating the entire run's worth of logical location array elements into each matched result, but now I see it's just a reference.